src/Authenticators/LDAPChangePasswordHandler.php

```diff
@@ -0,0 +1,163 @@
+            if (!$loginResult['success']) {
+                $form->clearMessage();
+                $form->sessionMessage(
+                    _t(
+                        'SilverStripe\\Security\\Member.ERRORPASSWORDNOTMATCH',
+                        'Your current password does not match, please try again'
+                    ),
+                    'bad'
+                );
+                // redirect back to the form, instead of using redirectBack() which could send the user elsewhere.
+                return $form->getController()->redirect($form->getController()->Link('changepassword'));
+            }
+        }
+
+        if (!$member) {
+            if ($this->getRequest()->getSession()->get('AutoLoginHash')) {
+                $member = Member::member_from_autologinhash($this->getRequest()->getSession()->get('AutoLoginHash'));
+            }
+
+            // The user is not logged in and no valid auto login hash is available
+            if (!$member) {
+                $this->getRequest()->getSession()->clear('AutoLoginHash');
+                return $form->getController()->redirect($form->getController()->Link('login'));
+            }
+        }
+
+        // Check the new password
+        if (empty($data['NewPassword1'])) {
+            $form->clearMessage();
+            $form->sessionMessage(
+                _t(
+                    'SilverStripe\\Security\\Member.EMPTYNEWPASSWORD',
+                    "The new password can't be empty, please try again"
+                ),
+                'bad'
+            );
+
+            // redirect back to the form, instead of using redirectBack() which could send the user elsewhere.
+            return $form->getController()->redirect($form->getController()->Link('changepassword'));
+        } elseif ($data['NewPassword1'] == $data['NewPassword2']) {
+            // Providing OldPassword to perform password _change_ operation. This will respect the
+            // password history policy. Unfortunately we cannot support password history policy on password _reset_
+            // at the moment, which means it will not be enforced on SilverStripe-driven email password reset.
+            $oldPassword = !empty($data['OldPassword']) ? $data['OldPassword']: null;
+
+            /** @var ValidationResult $validationResult */
+            $validationResult = $service->setPassword($member, $data['NewPassword1'], $oldPassword);
+
+            // try to catch connection and other errors that the ldap service can through
+            if ($validationResult->isValid()) {
+                Security::setCurrentUser($member);
+
+                $this->getRequest()->getSession()->clear('AutoLoginHash');
+
+                // Clear locked out status
+                $member->LockedOutUntil = null;
+                $member->FailedLoginCount = null;
+                $member->write();
+
+                if (!empty($_REQUEST['BackURL'])
```

Comment:
It might be better to use `$this->getRequest()->requestVar('BackURL')` to avoid using the superglobal